### PR TITLE
mgr/cephadm: raise, if crash keyring fails

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes: https://tracker.ceph.com/issues/44826
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -3296,6 +3296,8 @@ def _create_crash(self, daemon_id, host):
             'caps': ['mon', 'profile crash',
                      'mgr', 'profile crash'],
         })
+        if ret:
+            raise OrchestratorError(f'Failed to create crash keyring: {ret} "{err}"')
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py:3300]: There are many instances where the return code is not checked...

Should we create a wrapper that throws during failure: `mon_command_throws(...)` ?

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py:3300]: https://github.com/ceph/ceph/blob/9a3aecbaf3c42b48ba1dc1fdf670c3da01c14956/src/pybind/mgr/dashboard/services/ceph_service.py#L147-L183

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py:3300]: and actually, I really miss 

https://github.com/sebastian-philipp/ceph/blob/6fb514079204fed76e805b9a863704c523716598/src/python-common/ceph/mon_command_api/generated_api.py 

But I never managed to get something like this into the tree yet. I think @matthewoliver would also like to have it.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-04-30_13:24:45-rados-wip-kefu-testing-2020-04-30-1835-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-05-01_04:25:09-rados-wip-kefu-testing-2020-04-30-1912-distro-basic-smithi/

**Pull request merged**